x/auth/ante/sigverify.go

```diff
@@ -293,10 +301,19 @@ func (svd SigVerificationDecorator) verifySig(ctx sdk.Context, tx sdk.Tx, acc sd
 	genesis := ctx.BlockHeight() == 0
 	chainID := ctx.ChainID()
 	var accNum uint64
+	// if we are not in genesis use the account number from the account
 	if !genesis {
 		accNum = acc.GetAccountNumber()
 	}
 
+	// if the account number is 0 and the account is signing, the sign doc will not have an account number
+	if acc.GetSequence() == 0 && newlyCreated {
```

Comment:
Isn't the `acc.GetSequence() == 0` obsolete here because whenever `newlyCreated` is true, the sequence is 0?

Reply:
Yes, but it doesn't hurt in the grand scheme of things.